src/lib.rs

```diff
@@ -51,8 +51,8 @@ pub trait FileIO {
 pub trait Cache: Send + Sync {
   type Key: Hash + Eq;
   type Value;
-  async fn set<'a>(&'a self, key: Self::Key, value: Self::Value, ttl: NonZeroU64) -> anyhow::Result<Self::Value>;
-  async fn get<'a>(&'a self, key: &'a Self::Key) -> anyhow::Result<Self::Value>;
+  async fn set<'a>(&'a self, key: Self::Key, value: Self::Value, ttl: NonZeroU64) -> Option<Self::Value>;
+  async fn get<'a>(&'a self, key: &'a Self::Key) -> Option<Self::Value>;
```

Comment:
I think it's better if we specify the error types as an associated type. Error can happen during insertion and retrieval. For ex: the remote service, in case of cloudflare's KV store, might be inaccessible or offline.

Reply:
making Error an associated type will work but I think creating an Error type which will behave like an actual error for the get method and not an error in the set method doesn't seem right. Instead I am thinking I can just make the `V` type an `Option` and just simply return the `insert` and `get` results from the `TtlCache` with `Ok` everytime

Reply:
> making Error an associated type will work but I think creating an Error type which will behave like an actual error for the get method and not an error in the set method doesn't seem right. Instead I am thinking I can just make the `V` type an `Option` and just simply return the `insert` and `get` results from the `TtlCache` with `Ok` everytime

Actually this wouldn't work. I'll make an error type instead

Reply:
I changed the set method to output a `Result<()>`. Other ways just seemed to make things unncessarily complicated so I thought since we aren't using the old value from the cache when we do an insert, we can return unit type for now

Reply:
A `get` operation can fail or succeed with no value. I think a better design here would be `Result<Option<Self::Value>>`. Not having the value vs not being able to retrieve it are two separate things and that it should be propagated to the caller.

Reply:
So I will have to make this change in the `Cache` trait because the `Self::Value` type has to be the type that actually goes inside the `TtlCache`. If we're doing this then I think we can change the `set` to `Result<Option<Self::Value>>` as well